### ISSUE TITLE
[c] Invalid ik animation with 2 bones

### ISSUE THREAD

**Issue by kerrytazi** [EsotericSoftware/spine-runtimes#1742]
**Spine editor**
![spine-editor-leg](https://user-images.githubusercontent.com/24854851/90728302-57f3f400-e2cd-11ea-8c83-b97a4e184ae2.png)

**C runtime**
![c-runtime-leg](https://user-images.githubusercontent.com/24854851/90728352-70640e80-e2cd-11ea-9432-ac97b29bf221.png)

<details>
  <summary>json</summary>

```json
{
"skeleton": {
	"hash": "Ypp3F1a0LY9H55PSz6vCUqVbkG4",
	"spine": "3.8.97",
	"x": -405.21,
	"y": -70.51,
	"width": 683.78,
	"height": 383.2,
	"images": "./images/",
	"audio": "./audio/"
},
"bones": [
	{ "name": "root" },
	{ "name": "bone", "parent": "root", "length": 226.68, "rotation": 92.6, "x": 18.17, "y": 1024.29 },
	{ "name": "bone54", "parent": "bone", "length": 366.75, "rotation": 174.43, "x": 21.56, "y": -138.46 },
	{ "name": "bone55", "parent": "bone54", "length": 586.26, "rotation": 10.95, "x": 366.75 },
	{ "name": "bone56", "parent": "bone55", "length": 137.3, "rotation": -37.58, "x": 587.01, "y": 5.37 },
	{ "name": "bone71", "parent": "root", "x": 271.37, "y": 8.44 },
	{
		"name": "bone57",
		"parent": "bone71",
		"length": 155.38,
		"rotation": -167.91,
		"x": -115.86,
		"y": -22.53,
		"color": "ff3f00ff"
	},
	{ "name": "bone73", "parent": "bone71", "x": -54.77, "y": 97.82, "color": "ff3f00ff" }
],
"slots": [
	{ "name": "Florist_front_idle_part25", "bone": "root", "attachment": "Florist_front_idle_part25" }
],
"ik": [
	{
		"name": "bone57",
		"order": 1,
		"bones": [ "bone55", "bone56" ],
		"target": "bone57",
		"bendPositive": false
	},
	{
		"name": "bone73",
		"bones": [ "bone54", "bone55" ],
		"target": "bone73"
	}
],
"skins": [
	{
		"name": "default",
		"attachments": {
			"Florist_front_idle_part25": {
				"Florist_front_idle_part25": {
					"type": "mesh",
					"uvs": [ 0.7041, 0.24986, 0.77627, 0.30601, 0.79445, 0.35666, 0.81584, 0.41624, 0.84646, 0.39818, 0.88123, 0.37115, 0.89498, 0.36121, 0.96477, 0.27886, 0.97083, 0.30529, 0.99626, 0.47592, 1, 0.51828, 0.98659, 0.69793, 0.97511, 0.71044, 0.92696, 0.74265, 0.83977, 0.76448, 0.84059, 0.77905, 0.81745, 0.81711, 0.81558, 0.78588, 0.7721, 0.86995, 0.67526, 0.93379, 0.62832, 0.96167, 0.48057, 0.99584, 0.39183, 0.98882, 0.31622, 0.97675, 0.28234, 0.95478, 0.27237, 0.93649, 0.26598, 0.91827, 0.31798, 0.81549, 0.34638, 0.77243, 0.40036, 0.70656, 0.45068, 0.67147, 0.49479, 0.57775, 0.54955, 0.50099, 0.57587, 0.42351, 0.60073, 0.41395, 0.60552, 0.37627, 0.64235, 0.26634, 0.66924, 0.2497, 0.33936, 0.83983, 0.391, 0.86983, 0.46433, 0.86383, 0.5831, 0.79983, 0.71116, 0.68583, 0.79259, 0.59283, 0.87521, 0.51883, 0.95886, 0.47883, 0.72132, 0.60483, 0.78019, 0.45683, 0.78701, 0.40896 ],
					"triangles": [ 46, 41, 32, 16, 17, 15, 42, 41, 46, 34, 46, 32, 35, 46, 34, 46, 35, 36, 46, 36, 37, 17, 14, 15, 14, 17, 43, 17, 42, 43, 13, 14, 44, 14, 43, 44, 13, 44, 12, 11, 12, 45, 12, 44, 45, 10, 45, 9, 10, 11, 45, 42, 46, 43, 37, 47, 46, 46, 47, 43, 48, 47, 0, 0, 47, 37, 43, 3, 44, 3, 4, 44, 43, 47, 3, 3, 47, 48, 44, 6, 45, 44, 5, 6, 44, 4, 5, 9, 45, 8, 45, 6, 8, 48, 0, 1, 3, 48, 2, 48, 1, 2, 6, 7, 8, 22, 40, 21, 21, 41, 20, 21, 40, 41, 22, 23, 39, 23, 38, 39, 22, 39, 40, 23, 24, 38, 20, 41, 19, 24, 27, 38, 24, 25, 27, 25, 26, 27, 19, 42, 18, 19, 41, 42, 38, 28, 39, 39, 29, 40, 39, 28, 29, 29, 30, 40, 40, 30, 41, 38, 27, 28, 30, 31, 41, 31, 32, 41, 34, 32, 33, 18, 42, 17 ],
					"vertices": [ 3, 3, 562.69, -80.32, 0.256, 4, 32.99, -82.75, 0.53057, 6, -7.6, -132.92, 0.21343, 2, 4, 27.9, -45.14, 0.92852, 6, -39.07, -111.71, 0.07148, 2, 4, 34.61, -31.12, 0.95361, 6, -45.07, -97.37, 0.04639, 2, 4, 42.51, -14.63, 0.98915, 6, -52.13, -80.5, 0.01085, 2, 4, 31.17, -3.88, 0.99992, 6, -67.7, -81.82, 8.0E-5, 2, 4, 16.86, 7.51, 1, 6, -85.72, -84.92, 0, 2, 4, 11.37, 12.1, 1, 6, -92.8, -85.97, 0, 2, 4, -23.49, 31.46, 0.99999, 6, -130.45, -99.12, 1.0E-5, 2, 4, -19.16, 37.33, 0.99999, 6, -131.95, -91.99, 1.0E-5, 2, 4, 12.09, 69.36, 1, 6, -135.07, -47.35, 0, 2, 4, 20.47, 76.22, 1, 6, -134.62, -36.53, 0, 2, 4, 63.06, 92.89, 0.768, 6, -118.74, 6.37, 0.232, 2, 4, 68.57, 89.58, 0.75768, 6, -112.6, 8.28, 0.24232, 2, 4, 87.24, 73.16, 0.73805, 6, -87.92, 11.29, 0.26195, 2, 4, 113.04, 38.88, 0.70074, 6, -45.17, 7.76, 0.29926, 2, 4, 116.04, 41.04, 0.64037, 6, -44.78, 11.43, 0.35963, 2, 4, 129.95, 35.96, 0.5569, 6, -31.73, 18.44, 0.4431, 2, 4, 123.56, 31.28, 0.4556, 6, -32.49, 10.56, 0.5444, 2, 4, 152.46, 23.3, 0.34431, 6, -7.3, 26.82, 0.65569, 2, 4, 189.8, -9.84, 0.2295, 6, 42.27, 32.65, 0.7705, 2, 4, 207.22, -26.29, 0.11452, 6, 66.15, 34.72, 0.88548, 1, 6, 138.45, 28.03, 1, 1, 6, 180.42, 17.23, 1, 1, 6, 215.87, 6.53, 1, 1, 6, 230.87, -2.35, 1, 1, 6, 234.66, -7.88, 1, 1, 6, 236.75, -13.02, 1, 2, 3, 677.69, -286.72, 0.02844, 6, 206.51, -33.03, 0.97156, 2, 3, 668.87, -271.48, 0.08533, 6, 190.68, -40.73, 0.91467, 2, 3, 655.83, -243.14, 0.17067, 6, 161.44, -51.56, 0.82933, 2, 3, 650.74, -217.55, 0.256, 6, 135.59, -54.95, 0.744, 3, 3, 630.33, -192.95, 0.256, 4, 155.29, -130.74, 0.00163, 6, 109.59, -73.53, 0.74237, 3, 3, 614.89, -163.8, 0.256, 4, 125.27, -117.06, 0.1235, 6, 79.41, -86.85, 0.6205, 3, 3, 597.34, -148.37, 0.256, 4, 101.95, -115.54, 0.22252, 6, 62.76, -103.25, 0.52148, 3, 3, 596.63, -136.02, 0.256, 4, 93.87, -106.18, 0.29582, 6, 50.39, -103.06, 0.44818, 3, 3, 587.56, -132.39, 0.256, 4, 84.46, -108.84, 0.36424, 6, 46.11, -111.85, 0.37976, 3, 3, 562.62, -110.74, 0.256, 4, 51.49, -106.9, 0.4786, 6, 22.74, -135.17, 0.2654, 3, 3, 560.29, -97.16, 0.256, 4, 41.36, -97.56, 0.50139, 6, 9.02, -136.53, 0.24261, 1, 6, 197.59, -24.84, 1, 1, 6, 174.54, -12.17, 1, 2, 4, 225.32, -108.05, 0.13217, 6, 139.23, -6.15, 0.86783, 2, 4, 182.66, -65.62, 0.29955, 6, 79.18, -9.78, 0.70045, 2, 4, 126.81, -25.48, 0.48316, 6, 12.05, -24.77, 0.51684, 2, 4, 86.81, -2.51, 0.66317, 6, -31.71, -39.36, 0.33683, 2, 4, 50.68, 23.34, 0.8396, 6, -75.05, -49.15, 0.1604, 2, 4, 21.74, 53.85, 0.768, 6, -117.08, -50.45, 0.232, 2, 4, 106.62, -31.25, 0.96281, 6, 2.92, -43.69, 0.03719, 2, 4, 60, -24.7, 0.96314, 6, -32.98, -74.14, 0.03686, 2, 4, 47.86, -27.76, 0.96381, 6, -38.76, -85.24, 0.03619 ],
					"hull": 38,
					"edges": [ 0, 74, 0, 2, 6, 8, 8, 10, 10, 12, 12, 14, 14, 16, 16, 18, 18, 20, 20, 22, 22, 24, 24, 26, 26, 28, 28, 30, 30, 32, 32, 34, 34, 36, 36, 38, 38, 40, 40, 42, 42, 44, 44, 46, 46, 48, 48, 50, 50, 52, 52, 54, 54, 56, 56, 58, 58, 60, 60, 62, 62, 64, 64, 66, 66, 68, 68, 70, 70, 72, 72, 74, 76, 78, 78, 80, 80, 82, 82, 84, 84, 86, 86, 88, 88, 90, 92, 94, 2, 4, 4, 6, 4, 96, 96, 94, 6, 96 ],
					"width": 488,
					"height": 252
				}
			}
		}
	}
],
"animations": {
	"default": {
		"bones": {
			"bone73": {
				"rotate": [
					{ "curve": "stepped" },
					{ "time": 0.2667, "curve": "stepped" },
					{ "time": 0.6667, "curve": "stepped" },
					{ "time": 1.3333, "curve": "stepped" },
					{ "time": 1.6, "curve": "stepped" },
					{ "time": 2, "curve": "stepped" },
					{ "time": 2.6667 }
				],
				"translate": [
					{ "y": 69.38, "curve": 0.233, "c3": 0.613, "c4": 0.44 },
					{
						"time": 0.2667,
						"x": -24.95,
						"y": 13.52,
						"curve": 0.437,
						"c2": 0.37,
						"c3": 0.887,
						"c4": 0.87
					},
					{ "time": 0.6667, "x": 41.22, "y": 10.47, "curve": 0.25, "c3": 0.75 },
					{ "time": 0.7667, "x": 41.72, "y": 1.51, "curve": 0.25, "c3": 0.75 },
					{ "time": 0.9667, "x": 32.15, "y": -8.37, "curve": 0.25, "c3": 0.75 },
					{ "time": 1.3333, "y": 69.38, "curve": 0.233, "c3": 0.613, "c4": 0.44 },
					{
						"time": 1.6,
						"x": -24.95,
						"y": 13.52,
						"curve": 0.437,
						"c2": 0.37,
						"c3": 0.887,
						"c4": 0.87
					},
					{ "time": 2, "x": 41.22, "y": 10.47, "curve": 0.25, "c3": 0.75 },
					{ "time": 2.1, "x": 41.72, "y": 1.51, "curve": 0.25, "c3": 0.75 },
					{ "time": 2.3, "x": 32.15, "y": -8.37, "curve": 0.25, "c3": 0.75 },
					{ "time": 2.6667, "y": 69.38 }
				]
			},
			"bone57": {
				"rotate": [
					{ "curve": 0.264, "c3": 0.61, "c4": 0.38 },
					{ "time": 0.2667, "angle": 30.35, "curve": 0.305, "c2": 0.24, "c3": 0.661, "c4": 0.61 },
					{ "time": 0.6667, "curve": 0.264, "c3": 0.61, "c4": 0.38 },
					{ "time": 0.7667, "angle": 5.34, "curve": 0.472, "c2": 0.44, "c3": 0.872, "c4": 0.86 },
					{ "time": 1.3333, "curve": 0.264, "c3": 0.61, "c4": 0.38 },
					{ "time": 1.6, "angle": 30.35, "curve": 0.305, "c2": 0.24, "c3": 0.661, "c4": 0.61 },
					{ "time": 2, "curve": 0.264, "c3": 0.61, "c4": 0.38 },
					{ "time": 2.1, "angle": 5.34, "curve": 0.472, "c2": 0.44, "c3": 0.872, "c4": 0.86 },
					{ "time": 2.6667 }
				],
				"translate": [
					{ "x": 25.91, "curve": 0.264, "c3": 0.61, "c4": 0.38 },
					{ "time": 0.2667, "curve": "stepped" },
					{ "time": 0.6667 },
					{ "time": 1.3333, "x": 25.91, "curve": 0.264, "c3": 0.61, "c4": 0.38 },
					{ "time": 1.6, "curve": "stepped" },
					{ "time": 2 },
					{ "time": 2.6667, "x": 25.91 }
				]
			}
		}
	}
}
}```
</details>

**Comment by NathanSweet** [comment]
That fix looks right, matches spine-libgdx.

**Comment by badlogic** [comment]
Cheers, I've fixed this in both the 3.8 and 4.0-beta branch, and also fixed it in spine-cpp.
